database/factories/AppointmentFactory.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Database\Factories;
+
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Factories\Factory;
+use Lightit\Appointments\Domain\Models\Appointment;
+use Lightit\Clinics\Domain\Models\Clinic;
+use Lightit\Doctors\Domain\Models\Doctor;
+use Lightit\Users\Domain\Models\User;
+
+/**
+ * @extends Factory<Appointment>
+ */
+class AppointmentFactory extends Factory
+{
+    protected $model = Appointment::class;
+
+    public function definition(): array
+    {
+        return [
+            'start_time' => $start = fake()->dateTime(),
+            'end_time' => (clone $start)->modify('+' . fake()->numberBetween(15, 100) . 'minutes'),
+            'doctor_id' => Doctor::all()->random()->id,
+            'user_id' => User::all()->random()->id,
+            'clinic_id' => Clinic::all()->random()->id
```

Comment:
Whenever your model depends on another one/s, in the great majority of cases, you should let the factory create them

```suggestion
            'doctor_id' => DoctorFactory::new(),
            'user_id' => UserFactory::new(),
            'clinic_id' => ClinicFactory::new()
```